src/Model/batchprocessing/batchprocessingMachineLearning/MachineLearningTrainingStage.py

```diff
@@ -0,0 +1,411 @@
+        }
+
+    """
+    Below 3 Functions that helps identify
+    what perfomance calculation should be used for the Model
+    """
+
+    def cal_perfomance_gm(self,
+                          predictions):
+        return 'geometric mean', geometric_mean_score(self.test_label,
+                                                      predictions)
+
+    def cal_perfomance_f1_macro(self, predictions):
+        return 'f1_macro', f1_score(self.test_label,
+                                    predictions,
+                                    average='macro')
+
+    def cal_perfomance_accuracy(self, predictions):
+        return 'accuracy', accuracy_score(self.test_label, predictions)
+
+        # Check if Our Test Label is Balanced
+
+    def calculate_balance(self):
+
+        balance = True
+
+        for i in range(len(self.test_label.unique())):
+            if round(self.test_label.value_counts()[i] /
+                     len(self.test_label) * 100, 2) < 15:
+                balance = False
+
+        return balance
+
+    def castom_confusion_matrix(self,
+                                predictions):
+        unique_label = np.unique([self.test_label,
+                                  predictions])
+        cmtx = pd.DataFrame(
+            confusion_matrix(self.test_label,
+                             predictions,
+                             labels=unique_label),
+            index=['true:{:}'.format(x) for x in unique_label],
+            columns=['pred:{:}'.format(x) for x in unique_label]
+        )
+        return cmtx
+
+    """
+    Following function Tunnes and Runs All Classification Models
+    Then among all models it picks the best one
+    """
+
+    def classification_ml_tunned(self):
+        # Tunning Random Forest
+        param_grid_rf = [
+            {'n_estimators': [400, 500, 600],
+             'max_depth': [5, None],
+             'criterion': ['gini', 'entropy', 'log_loss']},
+            {'bootstrap': [False],
+             'n_estimators': [400, 500, 600],
+             'criterion': ['gini', 'entropy', 'log_loss'],
+             'max_depth': [5, None]}]
+
+        param_grid_mlp = [
+            {'hidden_layer_sizes': [(100,), (200,), (300,)],
+             'activation': ['identity', 'logistic',
+                            'tanh', 'relu'],
+             'solver': ['sgd', 'adam'],
+             'alpha': [0.01, 0.1],
+             'learning_rate_init': [0.01, 0.1]},
+            {'hidden_layer_sizes': [(100,), (200,), (300,)],
+             'activation': ['identity', 'logistic', 'tanh', 'relu'],
+             'solver': ['lbfgs'], 'alpha': [0.01, 0.1]}]
+
+        # Call Random Forest
+        forest_clas = RandomForestClassifier(random_state=42)
+        # Call MLP
+        mlp_cla = MLPClassifier(random_state=42, max_iter=5000)
+
+        # check if it is binary labels (2 classes)
+        if len(self.test_label.unique()) == 2:
+            print('running self.testLabel.unique()) == 2')
```

Comment:
A lot of print statements within the new/modified files (10 found, including this one), not sure if they're needed, otherwise remove

Reply:
Thanks. 
I have removed unneeded print statements